tests/certification/bindings/aws/s3/s3_test.go

```diff
@@ -0,0 +1,426 @@
+	}
+}
+
+func TestAWSS3CertificationTests(t *testing.T) {
+	defer teardown(t)
+
+	t.Run("S3SBasic", func(t *testing.T) {
+		S3SBasic(t)
+	})
+
+	t.Run("S3SForcePathStyle", func(t *testing.T) {
+		S3SForcePathStyle(t)
+	})
+
+	t.Run("S3SBase64", func(t *testing.T) {
+		S3SBase64(t)
+	})
+}
+
+// createObjectRequest is used to make a common binding request for create operation.
+func createObjectRequest(ctx flow.Context, client daprsdk.Client, dataBytes []byte, invokeCreateMetadata map[string]string) (*daprsdk.BindingEvent, error) {
+	invokeCreateRequest := &daprsdk.InvokeBindingRequest{
+		Name:      bindingsMetadataName,
+		Operation: "create",
+		Data:      dataBytes,
+		Metadata:  invokeCreateMetadata,
+	}
+
+	return client.InvokeBinding(ctx, invokeCreateRequest)
+}
+
+// listObejctRequest is used to make a common binding request for the list operation.
+func listObejctRequest(ctx flow.Context, client daprsdk.Client) (out *daprsdk.BindingEvent, err error) {
+	invokeRequest := &daprsdk.InvokeBindingRequest{
+		Name:      bindingsMetadataName,
+		Operation: "list",
+		Data:      nil,
+		Metadata:  nil,
+	}
+
+	out, invokeErr := client.InvokeBinding(ctx, invokeRequest)
+	if invokeErr != nil {
+		return nil, fmt.Errorf("%w", invokeErr)
+	}
+	return out, nil
+}
+
+// getObjectRequest is used to make a common binding request for the get operation.
+func getObjectRequest(ctx flow.Context, client daprsdk.Client, name string, isBase64 bool) (out *daprsdk.BindingEvent, err error) {
+	invokeGetMetadata := map[string]string{
+		"key":          name,
+		"encodeBase64": fmt.Sprintf("%t", isBase64),
+	}
+
+	return getObjectRequestWithMetadata(ctx, client, invokeGetMetadata)
+
+}
+
+// getObjectRequest is used to make a common binding request for the get operation passing metadata.
+func getObjectRequestWithMetadata(ctx flow.Context, client daprsdk.Client, invokeGetMetadata map[string]string) (out *daprsdk.BindingEvent, err error) {
+	invokeGetRequest := &daprsdk.InvokeBindingRequest{
+		Name:      bindingsMetadataName,
+		Operation: "get",
+		Data:      nil,
+		Metadata:  invokeGetMetadata,
+	}
+
+	out, invokeGetErr := client.InvokeBinding(ctx, invokeGetRequest)
+	if invokeGetErr != nil {
+		return nil, fmt.Errorf("%w", invokeGetErr)
+	}
+	return out, nil
+}
+
+// deleteObejctRequest is used to make a common binding request for the delete operation.
+func deleteObejctRequest(ctx flow.Context, client daprsdk.Client, name string) (out *daprsdk.BindingEvent, err error) {
+	invokeDeleteMetadata := map[string]string{
+		"key": name,
+	}
+
+	invokeGetRequest := &daprsdk.InvokeBindingRequest{
+		Name:      bindingsMetadataName,
+		Operation: "delete",
+		Data:      nil,
+		Metadata:  invokeDeleteMetadata,
+	}
+
+	out, invokeDeleteErr := client.InvokeBinding(ctx, invokeGetRequest)
+	if invokeDeleteErr != nil {
+		return nil, fmt.Errorf("%w", invokeDeleteErr)
+	}
+	return out, nil
+}
+
+// Verify S3 Basic Binding Support (Create, Get, List, Delete)
```

Comment:
What is the difference between these tests and the existing conformance tests?

Reply:
One difference I see is the fact the conformance tests require either `docker` or `cloud` resources to be ready prior testing.

Reply:
Is there any different in the test scenarios?

Reply:
For S3 Cert test in particular, it will test certain metadata tags that are not necessarily  common with other components like **forcePathStyle**, **encodeBase64**,  and **decodeBase64**.  I hope that answers your question.

Reply:
It does. Thanks.